docusaurus/docs/guides/data-flow-explainer.md

```diff
@@ -0,0 +1,162 @@
+handled by terminators.
+
+![image](/img/data-flow/terminators.png)
+
+There are three common ways in which this is done.
+
+### Static Terminators
+
+A static terminator can be created by an OpenZiti admin. The terminator will specify the address to
+dial. Statically created terminators will exist until they are deleted or their related service or
+router is deleted.
+
+### SDK Hosted
+
+When an SDK application hosts (or binds, in OpenZiti terms) a service, it will first connect to one
+or more edge routers. It can then request those edge routers to create terminators on its behalf.
+When the SDK app disconnects from the edge router, the terminator will automatically be cleaned up.
+
+The SDK application may handle incoming connections itself, or if it's acting as a reverse proxy, it
+may bridge the connection to another application server. The OpenZiti tunneler applications can do
+this and will use the `host.v1` or `host.v2` configurations to define how and where connections are
+forwarded to and what kind of health checks to run.
+
+### Edge Router Hosted
+
+Edge routers running in tunneler mode (ER/T) can also host services by acting as a reverse proxy.
+They will also use the `host. v1` and `host.v2` configurations, same as the SDK based tunnelers. An
+ER/T will create and delete terminators as services, configurations and policies are created,
+updated or deleted.
+
+## Client Connections
+
+When an SDK wants to connect to an OpenZiti hosted service, it will initiate the following sequence
+of events:
+
+1. Authenticate to the controller
+2. Create a service session for the desired service
+3. Connect to one or more edge routers
+4. Send a Dial request to the selected edge router
+5. The edge router will send a create circuit request to the controller
+6. The controller will find the least expensive path
+7. The controller will send route updates to non-terminal elements of the path
+8. The controller will instruct the terminating router to dial the service
+9. Once the circuit is established, data can begin to flow
+
+### Authentication
+
+As a first step, the SDK must be authenticated to the controller. The result of a successful
+authentication is an API Session. The SDK will use the API Session to make additional requests to
+the control and to authenticate with edge routers. In order for edge routers to be able to verify
+API Session from connecting SDKs, the controller must inform routers of new API Session as they are
+created.
+
+![image](/img/data-flow/client-api-session.png)
+
+An API Session can be extended, so a single API Session can generally be used for the lifetime of
+the application. A new API Session may be required if the application is unable to reach the
+controller for an extended period of time.
+
+### Create Service Session
+
+The SDK must now create service session. When creating a service session, the controller will verify
```

Comment:
Shall we maintain the distinction between API Session and Service Session by adopting the capitalization of both as a convention throughout the docs? If so, I believe it will clarify these similar terms, avoiding the ambiguity of "session" when the context doesn't make it obvious which proper type of session is in focus.

Reply:
I'm hesitant to do that as we don't use `Service Session` elsewhere. I  do use it to disambiguate it from API Session, as API Session and Service Session are easier than API Session and Session. But I'm worried about emphasizing it too much unless we also change it elsewhere, otherwise it might cause another kind of confusion. At least fabric sessions are now called circuits, so that's one less source of confusion :)

Reply:
I'd support and do legwork for clarifying [the current confusion](https://openziti.io/docs/learn/core-concepts/security/sessions/#session) around "Session" vs. "Service Session."

Reply:
Ok, I switched all the service sessions to Service Session.